src/Saunter/Serialization/NewtonsoftAsyncApiDocumentSerializer.cs

```diff
@@ -1,15 +1,31 @@
 ﻿using Newtonsoft.Json;
+using NJsonSchema;
+using NJsonSchema.Infrastructure;
 using Saunter.AsyncApiSchema.v2;
+using Saunter.Generation.SchemaGeneration;
+using System.Threading;
+using System.Threading.Tasks;
 
 namespace Saunter.Serialization
 {
     public class NewtonsoftAsyncApiDocumentSerializer : IAsyncApiDocumentSerializer
     {
         public string ContentType => "application/json";
 
-        public string Serialize(AsyncApiDocument document, AsyncApiOptions options)
+        public string Serialize(AsyncApiDocument document)
         {
-            return JsonConvert.SerializeObject(document, options.JsonSchemaGeneratorSettings.ActualSerializerSettings);
+            var contractResolver = JsonSchema.CreateJsonSerializerContractResolver(SchemaType.OpenApi3);
+            return JsonSchemaSerialization.ToJson(document, SchemaType.OpenApi3, contractResolver, Formatting.Indented);
+        }
```

Comment:
NJS has special handling for $ref so it provides these serialization hooks.

Btw: At some point I'll switch to System.Text.Json for all this, but this is quite some work and currently it's "transparent" in the sense that from a user perspective it doesnt really matter how the document is serialized/deserialized. 

Also document serialization is not a hot path so performance should not be an issue (ie STJ vs newtonsoft).

Reply:
Creating a custom serializer/resolver here makes sense, as you rightly point out, the "AsyncAPI document serialization settings" should be separate from the "Schema Generation settings".

However, I'm still not seeing the expected result.

Add an example to one of the `example/StreetlightsAPI.LightMeasuredEvent` properties
```c#
/// <example>123</example>
public int Lumens { get; set; }
```

Then you can still see `x-example` in the json payload
```json
"lightMeasuredEvent": {
    "properties": {
        "lumens": {
            "type": "integer",
            "x-example": 123
        },
```

I believe this is due to the use of `JsonReferenceConverter`
```c#
[JsonConverter(typeof(JsonReferenceConverter))]
public class AsyncApiDocument : ICloneable
{}
```

Which does not respect the `contractResolver`

https://github.com/RicoSuter/NJsonSchema/blob/master/src/NJsonSchema/Converters/JsonReferenceConverter.cs#L55